pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -298,3 +312,29 @@ func (r *ReconcileMongoDB) createFromYaml(instance *operatorv1alpha1.MongoDB, ya
 
 	return nil
 }
+
+func (r *ReconcileMongoDB) getstorageclass() (string, error) {
+	scList := &storagev1.StorageClassList{}
+	err := r.client.List(context.TODO(), scList)
+	if err != nil {
+		return "", err
+	}
+	if len(scList.Items) == 0 {
+		return "", fmt.Errorf("could not storage class in the cluster")
+	}
+	for _, sc := range scList.Items {
```

Comment:
can we use a single loop to get all the sc and then choose one? We can later reuse this code if there is strategy change.

```
    var defaultSC []string
    var nonDefaultSC []string
    for _, sc := range scList.Items {
        if sc.Provisioner == "kubernetes.io/no-provisioner" {
            continue
        }
        if sc.ObjectMeta.GetAnnotations()["storageclass.beta.kubernetes.io/is-default-class"] == "true" {
            defaultSC = append(defaultSC, sc.GetName())
            continue
        }
        nonDefaultSC = append(nonDefaultSC, sc.GetName())
    }

    if len(defaultSC) != 0 {
        return defaultSC[1], nil
    }

    if len(nonDefaultSC) != 0 {
        return nonDefaultSC[1], nil
    }
```